quests/rl/trainer/model.py

```diff
@@ -14,74 +14,36 @@
-        network, the loss function is altered to reduce the difference between
-        predicted Q-values and Target Q-values.
-
-        Args:
-            state_size (int): the number of parameters used to define the state
-                the agent is in.
-            action_size (int): the number of different actions the agent can
-                take.
-            learning_rated (float): the nueral network's learning rate.
-            hidden_neurons (int): the number of neurons to use per hidden
-                layer.
-            name (str): What to call the network, as shown in tensorboard.
-        """
-        with tf.variable_scope(name):
-            self.states = tf.placeholder(
-                tf.float32, shape=((None,) + state_size), name='states')
-            self.actions = tf.placeholder(
-                tf.int32, shape=(None), name='actions')
-
-            # target_Q is R(s,a) + ymax Qhat(s', a')
-            self.target_Q = tf.placeholder(tf.float32, [None], name="target")
-
-            # TODO(ddetering): Switch to keras.
-            self.fully_connected_1 = tf.layers.dense(
-                inputs=self.states,
-                units=hidden_neurons,
-                activation=tf.nn.relu,
-                name="fully_connected_1")
-
-            self.fully_connected_2 = tf.layers.dense(
-                inputs=self.fully_connected_1,
-                units=hidden_neurons,
-                activation=tf.nn.relu,
-                name="fully_connected_2")
-
-            self.output = tf.layers.dense(
-                inputs=self.fully_connected_2,
-                units=action_size,
-                activation=None)
-
-            # Hot encode and multiply across to select Q value for an action.
-            self.actions_hot = tf.one_hot(self.actions, action_size)
-            # Q is our predicted Q value.
-            self.Q = tf.reduce_sum(
-                tf.multiply(self.output, self.actions_hot), axis=1)
-
-            # The loss is the difference between predicted Q_values and
-            # the Q_target Sum(Qtarget - Q)^2
-            self.loss = tf.reduce_mean(tf.square(self.target_Q - self.Q))
-            self.optimizer = (
-                tf.train.RMSPropOptimizer(learning_rate).minimize(self.loss))
-            tf.summary.scalar("loss", self.loss)  # Track in Tensorboard.
+def deep_q_network(state_shape, action_size, learning_rate, hidden_neurons):
+    """Creates a Deep Q Network to emulate Q-learning.
+
+    Creates a two hidden-layer Deep Q Network. Similar to a typical nueral
+    network, the loss function is altered to reduce the difference between
+    predicted Q-values and Target Q-values.
+
+    Args:
+        space_shapes ():
+        learning_rate (float): the nueral network's learning rate.
+        hidden_neurons (int): the number of neurons to use per hidden
+            layer.
+    """
+    model = tf.keras.Sequential([
+        tf.keras.layers.Dense(
```

Comment:
nit(Optional): Perhaps number of layers can also be a parameter.

Reply:
I'll add it as an exercise for the reader :)